public/app/features/plugins/admin/components/InstallControls/InstallControlsButton.tsx

```diff
@@ -0,0 +1,93 @@
+import React from 'react';
+import { AppEvents } from '@grafana/data';
+import { Button, HorizontalGroup, useStyles2 } from '@grafana/ui';
+import appEvents from 'app/core/app_events';
+import { api } from '../../api';
+import { ActionTypes, CatalogPlugin } from '../../types';
+import { getStyles } from './index';
+
+type InstallControlsButtonProps = {
+  disabled: boolean;
+  hasInstalledPanel: boolean;
+  dispatch: React.Dispatch<any>;
+  plugin: CatalogPlugin;
+  pluginStatus: 'install' | 'update' | 'uninstall';
+};
+
+export function InstallControlsButton({
+  disabled,
+  dispatch,
+  plugin,
+  pluginStatus,
+  hasInstalledPanel,
+}: InstallControlsButtonProps) {
+  const uninstallBtnText = disabled ? 'Uninstalling' : 'Uninstall';
+  const updateBtnText = disabled ? 'Updating' : 'Update';
+  const installBtnText = disabled ? 'Installing' : 'Install';
```

Comment:
**Question:** should `disabled` only be used when the action is "in progress"?

Maybe it would make sense to rename the prop to something like `isInProgress` to reveal more what it is intended to be used for?